nix/cargo-vendor.nix

```diff
@@ -0,0 +1,27 @@
+# patched version of cargo to support `cargo vendor` for vendoring dependencies
```

Comment:
Can we move it to `etc/nix` to not overly pollute the top-level scope, and also add a `README.md` for the directory so that people are aware of why we need it and where is it used?

Reply:
yes, will do

Reply:
done